tests/test_structures/test_datasample.py

```diff
@@ -156,3 +157,20 @@ def test_set_pred_score(self):
         data_sample = ClsDataSample()
         data_sample.set_pred_score(torch.tensor([0.1, 0.1, 0.6, 0.1, 0.1]))
         self.assertEqual(data_sample.pred_label.num_classes, 5)
+
+
+class TestMultiTaskDataSample(GenericTest):
```

Comment:
please add the following tests : 

### to_target_data_sample

* `data_sample.to_target_data_sample('ClsDataSample', 'task0')`
  * if gt_label['task0'] is a number with 'ClsDataSample' -> return a ClsDataSample
  * if gt_label['task0'] is a number bigger than 'num_data_sample' with 'ClsDataSample'  with MultiTaskDataSample(metainfo={ 'task0': {num_classes='10'}, 'task1': {} })-> throw an error
  * if gt_label['task0'] is a string -> throw an error
  * if 'task0' is not in gt_label -> throw an error
 
* `data_sample.to_target_data_sample('MultiTaskDataSample', 'nested_task')` -> return an other  'MultiTaskDataSample' with `get_gt_label` as expected
* `data_sample.to_target_data_sample('MultiTaskDataSample', 'nested_task')` throw an error if the gt_label for key `nested_task` is a number (and not a dictonnary)